cobbler/items/network_interface.py

```diff
@@ -0,0 +1,896 @@
+"""
+All code belonging to network interfaces
+
+
+Changelog (NetworkInterface):
+
+V3.4.0 (unreleased):
+    * Changes:
+        * Constructor: ``kwargs`` can now be used to seed the item during creation.
+        * ``virt_type``: str - Inheritable; One of "qemu", "kvm", "xenpv", "xenfv", "vmware", "vmwarew", "openvz" or
+          "auto".
+V3.3.4 (unreleased):
+    * No changes
+V3.3.3:
+    * Changed:
+        * ``to_dict()``: Accepts new parameter ``resolved``
+        * ``virt_bridge``: Can now be set to ``<<inherit>>`` to get its value from the settings key
+          ``default_virt_bridge``
+V3.3.2:
+    * No changes
+V3.3.1:
+    * No changes
+V3.3.0:
+    * This release switched from pure attributes to properties (getters/setters).
+    * Added:
+        * ``NetworkInterface`` is now a class.
+        * Serialization still happens inside the system collection.
+        * Properties have been used.
+V3.2.2:
+    * No changes
+V3.2.1:
+    * No changes
+V3.2.0:
+    * No changes
+V3.1.2:
+    * No changes
+V3.1.1:
+    * No changes
+V3.1.0:
+    * No changes
+V3.0.1:
+    * No changes
+V3.0.0:
+    * Field definitions now split of ``System`` class
+V2.8.5:
+    * Initial tracking of changes for the changelog.
+    * Field definitions part of ``System`` class
+    * Added:
+        * ``mac_address``: str
+        * ``connected_mode``: bool
+        * ``mtu``: str
+        * ``ip_address``: str
+        * ``interface_type``: str - One of "na", "bond", "bond_slave", "bridge", bridge_slave", "bonded_bridge_slave",
+          "infiniband"
+        * ``interface_master``: str
+        * ``bonding_opts``: str
+        * ``bridge_opts``: str
+        * ``management``: bool
+        * ``static``: bool
+        * ``netmask``: str
+        * ``if_gateway``: str
+        * ``dhcp_tag``: str
+        * ``dns_name``: str
+        * ``static_routes``: List[str]
+        * ``virt_bridge``: str
+        * ``ipv6_address``: str
+        * ``ipv6_prefix``: str
+        * ``ipv6_secondaries``: List[str]
+        * ``ipv6_mtu``: str
+        * ``ipv6_static_routes``: List[str]
+        * ``ipv6_default_gateway``: str
+        * ``cnames``: List[str]
+"""
+import enum
+import logging
+from ipaddress import AddressValueError
+from typing import TYPE_CHECKING, Any, Dict, List, Union
+
+from cobbler import enums, utils, validate
+from cobbler.decorator import InheritableProperty
+
+if TYPE_CHECKING:
+    from cobbler.api import CobblerAPI
+
+
+class NetworkInterface:
+    """
+    A subobject of a Cobbler System which represents the network interfaces
+    """
+
+    def __init__(
+        self, api: "CobblerAPI", system_name: str, *args: Any, **kwargs: Any
```

Comment:
Should we add `# pylint: disable=unused-argument` (or `# pylint: disable-next=unused-argument` in the line above this)?

Reply:
Very reasonable. Thanks for the tip.